### PR TITLE
Unblock the accept thread on shutdown

### PR DESCRIPTION
**Pull request by moises-silva**
I'd like to find a better way, but, this works and is not completely awful, is it? :)

If this is not acceptable I'd like to initiate a discussion on ideas on how to handle shutdown correctly. Currently if tiny-http is used in a loadable module (I am using it inside a project that can load/unload modules from an interactive command line a system administrator controls), when the server goes out of scope the socket lingers because the thread never terminates unless there's a new tcp connection received (accept is a blocking call).

An alternative to this would be to put the socket in non-blocking mode and then somehow poll() or select() on it with a timeout to check every sec for the shutdown flag (close trigger bool).

I may be wrong, but I don't see many downsides to this implementation. The Drop() call *could* hang for a bit on unusual situations like the server being inaccessible to itself (e.g 127 -> 127 firewall blocking or whatever), but the connection will timeout and move on. Most cases are better off with this as far as I can tell.

Thoughts?

### CODE REVIEW

**Comment by frewsxcv** [comment]
This sounds good to me! Looks like there are merge conflicts though

File: src/lib.rs [hunk 1]
```diff
@@ -411,7 +410,13 @@ impl<'a> Iterator for IncomingRequests<'a> {
 }
 
 impl Drop for Server {
+    #[allow(unused_must_use)]
     fn drop(&mut self) {
         self.close.store(true, Relaxed);
+        // Connect briefly to ourselves to unblock the accept thread
+        let maybe_stream = TcpStream::connect(self.listening_addr);
+        if let Ok(stream) = maybe_stream {
+            stream.shutdown(Shutdown::Both);
```
Comment (frewsxcv) [inline src/lib.rs]: This can be `let _ = stream.shutdown(Shutdown::Both);`, then we can get rid of `#[allow(unused_must_use)]`

Comment (moises-silva) [inline src/lib.rs]: Ah! right, thanks, much better. Done.

**Comment by frewsxcv** [comment]
Thanks!

@aelita-mergebot r+

**Comment by frewsxcv** [comment]
@aelita-mergebot r+

**Pull request merged**